src/api/cpp/array.cpp

```diff
@@ -471,21 +471,31 @@ array::array_proxy &af::array::array_proxy::operator=(const array &other) {
         }
     }
 
-    af_array par_arr = nullptr;
+    af_array par_arr = 0;
 
+    dim4 dims = impl->parent_->dims();
     if (impl->is_linear_) {
         AF_THROW(af_flat(&par_arr, impl->parent_->get()));
+        // af_flat creates a new reference, remove the original reference
+        // in order to prevent triggering a memcpy in af_assign_gen when refCount > 1
+        af_array empty = 0;
+        impl->parent_->set(empty);
         nd = 1;
     } else {
         par_arr = impl->parent_->get();
     }
 
-    af_array tmp = nullptr;
+    af_array tmp = 0;
```

Comment:
I know we have this all over the place in this code but can you call this `flat_result` or something more descriptive?

Reply:
Done